app/containers/DatasetPage.js

```diff
@@ -102,6 +123,40 @@ class DatasetPage extends Component {
     )
   }
 
+  handleCreateDatasetUpload = () => {
+    if(this.state.name === "") {
+      alert("Please set a name for the dataset =)")
```

Comment:
Since react textfield's `required` attribute will work only when it is involved in a `<form>` with a "submit" button, or when the button knows which element to check validity, I applied the latter solution by using React.ref.

Reply:
I used `inputRef` in the textField and make it a `required` one, like this: https://codesandbox.io/s/empty-night-kfgpe?file=/src/App.js:196-204. The difference is I used `React.createRef()` in the constructor, instead of the ref initialization in the link.